src/main/java/redis/clients/jedis/providers/ClusterConnectionProvider.java

```diff
@@ -158,6 +165,25 @@ public Connection getConnectionFromSlot(int slot) {
     }
   }
 
+  public Connection getReplicaConnectionFromSlot(int slot) {
+    List<ConnectionPool> connectionPools = cache.getSlotReplicaPools(slot);
+    ThreadLocalRandom random = ThreadLocalRandom.current();
+    if (connectionPools != null && !connectionPools.isEmpty()) {
+      // pick up randomly a connection
+      int idx = random.nextInt(connectionPools.size());
+      return connectionPools.get(idx).getResource();
+    }
+
+    renewSlotCache();
+    connectionPools = cache.getSlotReplicaPools(slot);
+    if (connectionPools != null && !connectionPools.isEmpty()) {
+      int idx = random.nextInt(connectionPools.size());
+      return connectionPools.get(idx).getResource();
+    }
+
+    return getConnection();
```

Comment:
Not sure about `renewSlotCache()` here right now. Let's keep that responsibility for `getConnectionFromSlot(slot)`.
```suggestion
    renewSlotCache();
    connectionPools = cache.getSlotReplicaPools(slot);
    if (connectionPools != null && !connectionPools.isEmpty()) {
      int idx = random.nextInt(connectionPools.size());
      return connectionPools.get(idx).getResource();
    }

    return getConnectionFromSlot(slot);
```